2-mandatory/3-shopping-list.js

```diff
@@ -19,7 +19,14 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+  let shoppingList = {};
+  shoppingList.name = recipe.name;
+  shoppingList.items = [];
+  for (let element of recipe.ingredients){
+    if(!(pantry.fridgeContents.includes(element) || pantry.cupboardContents.includes(element)))
+    {shoppingList.items.push(element)}
+  };
+  return shoppingList;
 }
 
```

Comment:
For extra practice - you could try writing the above exercise using the Array `filter` method.